### PR TITLE
Try some GitHub Actions Workflow Commands

### PR DESCRIPTION
**Pull request by reneme**
I felt like playing with GitHub Actions Commands. This is meant as a demo/test bed. If we want to keep it, I could split it into individual PRs.

See here for details: https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions

### Grouping the Build Log

That change is simple but potentially simplifies the log exploration significantly. The GitHub Actions specific stuff is printed only when `export $GITHUB_ACTIONS=true` (as [stated in GitHub's documentation](https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables)):

> Always set to true when GitHub Actions is running the workflow. You can use this variable to differentiate when tests are being run locally or by GitHub Actions.

image

### Clang Tidy Warnings in the Pull Request

`clang-tidy` allows exporting its findings as a YAML file. I added a small python helper that reads this and translates it into GitHub Actions annotations with file coordinates.

image

### CODE REVIEW

**Comment by reneme** [comment]
Other additions worth looking into:

* Translate the JUnit results into GitHub Actions Commands
* Visualize compiler warnings/errors similarly to clang-tidy
   (off-the-shelf [problem matchers](https://michaelheap.com/getting-started-problem-matchers/) exist for this)
* ...

File: src/scripts/ci/gh_clang_tidy_fixes_in_pr.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+            if "Diagnostics" not in fixes:
+                raise RuntimeError(f"No Diagnostics found in {yaml_file}")
+            return [Diagnostic(diag) for diag in fixes["Diagnostics"]]
+
+
+    def __map_file_path(self, file_path, base_path): # pylint: disable=unused-argument
+        if file_path.endswith(".h"):
+            raise RuntimeError(f"Header file {file_path} cannot be handled")
+        # TODO: try to map include files (residing in build/include) onto their
+        #       origin path in src/lib etc.
```
Comment (reneme) [inline src/scripts/ci/gh_clang_tidy_fixes_in_pr.py:42]: This might become urgent once https://github.com/randombit/botan/issues/3644 is done.

**Review by reneme — COMMENTED** [review]
No need that this was still a draft. @randombit both CI features should be ready to merge, if you think they could be useful.

**Review by randombit — APPROVED** [review]
:+1:

**Pull request merged**